dotcom-rendering/src/model/buildLightboxImages.ts

```diff
@@ -0,0 +1,187 @@
+import { getLargest, getMaster } from '../components/ImageComponent';
+import type {
+	FEElement,
+	ImageBlockElement,
+	ImageForLightbox,
+} from '../types/content';
+
+type Orientation = 'horizontal' | 'portrait';
+
+/**
+ * Only allow the lightbox to show images that have a source with a width greater
+ * than 620 pixels.
+ *
+ * Frontend has similar logic here:
+ * https://github.com/guardian/frontend/blob/126dfcbc1aa961650b7f7ff41ee50a12782bb62e/common/app/model/content.scala#L549
+ *
+ */
+const isLightboxable = (
+	element: ImageBlockElement,
+	orientation: Orientation,
+): boolean => {
+	switch (orientation) {
+		case 'horizontal':
+			// If any width is above 620 we allow this image in lightbox
+			return element.media.allImages.some(
+				(mediaImg) => parseInt(mediaImg.fields.width, 10) > 620,
+			);
+		case 'portrait':
+			// If any height is above 620 we allow this image in lightbox
+			return element.media.allImages.some(
+				(mediaImg) => parseInt(mediaImg.fields.height, 10) > 620,
+			);
+	}
+};
+
+/**
+ * Older legacy images use a different url format so we need to use a different
+ * approach when extracting the id of the image
+ */
+const decideImageId = (image: ImageForLightbox): string | undefined => {
+	const url = new URL(image.masterUrl);
+	switch (url.hostname) {
+		case 'media.guim.co.uk': {
+			// E.g.
+			// https://media.guim.co.uk/be634f340e477a975c7352f289c4353105ba9e67/288_121_3702_2221/140.jpg
+			// This bit                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+			return url.pathname.split('/')[1];
+		}
+		case 'i.guim.co.uk':
+		case 'uploads.guim.co.uk':
+		case 'static-secure.guim.co.uk':
+		case 'static.guim.co.uk':
+			// E.g.
+			// https://static.guim.co.uk/sys-images/Guardian/Pix/pictures/2015/5/26/1432666797165/59de49e2-553f-4b52-b7ac-09bda7f63e4b-220x132.jpeg
+			// This bit                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+			return url.pathname.split('/').reverse()[0];
+		default:
+			return;
+	}
+};
+
+const buildLightboxImage = (
+	element: ImageBlockElement,
+): ImageForLightbox | undefined => {
+	const masterImage =
+		getMaster(element.media.allImages) ??
+		getLargest(element.media.allImages);
+
+	// Rare, but legacy content might not have a url that we can use with Fastly
+	// so we can't include them in lightbox
+	if (!masterImage) return;
+
+	const width = parseInt(masterImage.fields.width, 10);
+	const height = parseInt(masterImage.fields.height, 10);
+
+	const orientation = width >= height ? 'horizontal' : 'portrait';
```

Comment:
Isn't orientation “landscape” rather than “horizontal”? Could help if you exported the `Orientation` type.